src/ethernet/RsRtspClient.cpp

```diff
@@ -185,30 +185,34 @@ int RsRTSPClient::stop()
 
 int RsRTSPClient::close()
 {
-    unsigned res = this->sendTeardownCommand(*this->m_scs.m_session, this->continueAfterTEARDOWN);
-    // wait for continueAfterTEARDOWN to finish
-    std::unique_lock<std::mutex> lck(m_commandMtx);
-    m_cv.wait_for(lck, std::chrono::seconds(RTSP_CLIENT_COMMANDS_TIMEOUT_SEC), [this] { return m_commandDone; });
-    // for the next command
-    if (!m_commandDone)
     {
-        RsRtspReturnValue err = {RsRtspReturnCode::ERROR_TIME_OUT, "client time out"};
-        throw std::runtime_error(format_error_msg(__FUNCTION__, err));
-    }
-    m_commandDone = false;
+        unsigned res = this->sendTeardownCommand(*this->m_scs.m_session, this->continueAfterTEARDOWN);
+        // wait for continueAfterTEARDOWN to finish
+        std::unique_lock<std::mutex> lck(m_commandMtx);
+        m_cv.wait_for(lck, std::chrono::seconds(RTSP_CLIENT_COMMANDS_TIMEOUT_SEC), [this] { return m_commandDone; });
+        // for the next command
+        if (!m_commandDone)
+        {
+            RsRtspReturnValue err = {RsRtspReturnCode::ERROR_TIME_OUT, "client time out"};
+            throw std::runtime_error(format_error_msg(__FUNCTION__, err));
+        }
+        m_commandDone = false;
 
-    if (m_lastReturnValue.exit_code != RsRtspReturnCode::OK)
-    {
-        throw std::runtime_error(format_error_msg(__FUNCTION__, m_lastReturnValue));
+        if (m_lastReturnValue.exit_code != RsRtspReturnCode::OK)
+        {
+            throw std::runtime_error(format_error_msg(__FUNCTION__, m_lastReturnValue));
+        }
     }
     m_eventLoopWatchVariable = ~0;
-    std::unique_lock<std::mutex> lk(m_taskSchedulerMutex);
+    {
+        std::lock_guard<std::mutex> lk(m_taskSchedulerMutex);
+    }
```

Comment:
I don't understand as the Lock_guard has no effect here.
Did you mean to move the var assignment into the scoped range ?